homeassistant/components/tessie/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""Tessie integration."""
+from http import HTTPStatus
+
+from aiohttp import ClientError, ClientResponseError
+from tessie_api import get_state_of_all_vehicles
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Tessie config."""
+    api_key = entry.data[CONF_ACCESS_TOKEN]
+
+    try:
+        vehicles = await get_state_of_all_vehicles(
+            session=async_get_clientsession(hass),
+            api_key=api_key,
+            only_active=True,
+        )
+    except ClientResponseError as e:
+        if e.status == HTTPStatus.UNAUTHORIZED:
+            raise ConfigEntryAuthFailed from e
+        raise ConfigEntryNotReady from e
+    except ClientError as e:
+        raise ConfigEntryNotReady from e
```

Comment:
It would be nicer if the library raised named exceptions for authentication errors instead of having to capture aiohttp exceptions

Reply:
It would, but it doesnt, and its not my library.